WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift

```diff
@@ -268,6 +272,16 @@ class GutenbergViewController: UIViewController, PostEditor {
 
         gutenberg.delegate = self
         showInformativeDialogIfNecessary()
+
+        let themeSupportStore = StoreContainer.shared.editorTheme
+        themeSupportQuery = themeSupportStore.query(EditorThemeQuery(blog: post.blog))
+        themeSupportReceipt = themeSupportStore.onStateChange { [weak self] (_, state) in
+            DispatchQueue.main.async {
+                if let strongSelf = self, let themeSupport = state.editorTheme(forBlog: strongSelf.post.blog)?.themeSupport {
+                    strongSelf.gutenberg.updateTheme(themeSupport)
+                }
+            }
+        }
```

Comment:
I've considered moving these calls outside of the GutenbergViewController. Ultimately I liked it here better because:

- It's closer to the location where it is being used.
- It's a fast enough call to where the average case will have this loaded before the user sees the editor. 
- If on a significantly slow network and a user navigates quickly to the editor, then there would still need to be some hook to the Gutenberg editor to call update theme.
- This only matters for the initial load of the editor for a blog, or if the theme changes; because the theme will load from a cache if no changes were detected.

Con: 
- There is a hiccup where the event can fire while the editor is still loading. I put a workaround for that issue [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/14085/files#diff-ae5d7f8fb9cae1bc2a433c8133c06674R578).

Reply:
Can we move this code to an extension file: `GutenbergViewController+ThemeSupport`?
And have all the methods associated with theme support moved there?

Reply:
Yeah that sounds like a good plan.

Reply:
I ended up moving these to an extension in the same file with a comment. Moving it to a new file would have resulted in changing the access level to a few objects like `gutenberg` and the `themeSupportQuery` and `themeSupportReceipt` which I feel are better left as `private`

Let me know if this works for you.